### PR TITLE
feat: add v2 split provider metadata

### PR DESCRIPTION
**Pull request by r0ohafza**

### CODE REVIEW

File: packages/splits-sdk/src/client/splitV2.ts [hunk 1]
```diff
@@ -621,6 +623,52 @@ export class SplitV2Client extends SplitV2Transactions {
...
+
+    if (!updateLogs || updateLogs.length == 0) return { split }
+
+    updateLogs.sort((a, b) => (a.blockNumber > b.blockNumber ? -1 : 1))
+
+    split.recipients = updateLogs[0].args._split.recipients as Address[]
+    split.allocations = updateLogs[0].args._split.allocations as bigint[]
+    split.totalAllocation = updateLogs[0].args._split.totalAllocation
+    split.distributionIncentive =
+      updateLogs[0].args._split.distributionIncentive
```
Comment (mihoward21) [inline packages/splits-sdk/src/client/splitV2.ts:668]: I think this is right but just double checking. The UpdateSplit event will emit the full list of addresses/allocations, right? That's why just using the most recent one works, right?

File: packages/splits-sdk/src/client/splitV2.ts [hunk 2]
```diff
@@ -621,6 +623,52 @@ export class SplitV2Client extends SplitV2Transactions {
...
+    this._requirePublicClient()
+
+    const [updateLogs, createLogs, owner] = await Promise.all([
+      this._publicClient?.getLogs({
+        address: splitAddress,
+        event: splitUpdatedEvent,
+        strict: true,
+      }),
+      this._publicClient?.getLogs({
+        event: splitCreatedEvent,
```
Comment (mihoward21) [inline packages/splits-sdk/src/client/splitV2.ts:637]: this event comes from the factory, right? should we be setting the factory as the `address`? I see your filtering on the args, just not sure if filtering on the `address` is more efficient/better for any reason.

File: packages/splits-sdk/src/client/splitV2.ts (hunk 1 above)
Comment (r0ohafza) [reply packages/splits-sdk/src/client/splitV2.ts:668]: yes

File: packages/splits-sdk/src/client/splitV2.ts [hunk 3]
```diff
@@ -621,6 +623,52 @@ export class SplitV2Client extends SplitV2Transactions {
...
+      distributionIncentive:
+        createLogs[0].args.splitParams.distributionIncentive,
+      type: getSplitType(this._chainId, createLogs[0].address),
+      owner,
+      creator: createLogs[0].args.creator,
+    }
+
+    if (!updateLogs || updateLogs.length == 0) return { split }
+
+    updateLogs.sort((a, b) => (a.blockNumber > b.blockNumber ? -1 : 1))
```
[the remark this answers is not in the document]
Comment (r0ohafza) [reply packages/splits-sdk/src/client/splitV2.ts:662]: good catch, yeah that makes sense

File: packages/splits-sdk/src/client/splitV2.ts [hunk 4]
```diff
@@ -621,6 +623,52 @@ export class SplitV2Client extends SplitV2Transactions {
   async owner(split: Address): Promise<Address> {
     return this._getSplitV2Contract(split).read.owner()
   }
+
+  async getSplitMetadata(splitAddress: Address): Promise<{ split: SplitV2 }> {
+    this._requirePublicClient()
+
+    const [updateLogs, createLogs, owner] = await Promise.all([
+      this._publicClient?.getLogs({
```
Comment (mihoward21) [inline packages/splits-sdk/src/client/splitV2.ts:631]: we'll want to set the start block to the factory deployment block (for both `getLogs` calls)

File: packages/splits-sdk/src/client/splitV2.ts (hunk 2 above)
Comment (r0ohafza) [reply packages/splits-sdk/src/client/splitV2.ts:637]: yeah, adding the factory as the address makes sense

File: packages/splits-sdk/src/client/splitV2.ts [hunk 5]
```diff
@@ -621,6 +623,52 @@ export class SplitV2Client extends SplitV2Transactions {
   async owner(split: Address): Promise<Address> {
     return this._getSplitV2Contract(split).read.owner()
   }
+
+  async getSplitMetadata(splitAddress: Address): Promise<{ split: SplitV2 }> {
```
Comment (mihoward21) [inline packages/splits-sdk/src/client/splitV2.ts:627]: let's just test this with a couple different providers (not sure who the biggest are, alchemy/infura/quicknode?) curious what performance looks like / if it breaks for any. i know a while ago i was having trouble fetching more than 10k logs in a single request. maybe providers handle that better now, or maybe viem handles it under the hood for us, or maybe we'll need to custom handle that ourselves.

**Review by mihoward21 — COMMENTED** [review]
just a few small things

File: packages/splits-sdk/src/client/splitV2.ts (hunk 5 above)
Comment (r0ohafza) [reply packages/splits-sdk/src/client/splitV2.ts:627]: yeah, I am not sure what the performance looks like rn, will test after I do some deployments

File: packages/splits-sdk/src/client/splitV2.ts [hunk 6]
```diff
@@ -627,24 +628,31 @@ export class SplitV2Client extends SplitV2Transactions {
...
       this.owner(splitAddress),
     ])
 
     if (!createLogs) throw new Error('Split not found')
 
+    const updateLogs = await this._publicClient?.getLogs({
+      address: splitAddress,
+      event: splitUpdatedEvent,
+      strict: true,
+      fromBlock: createLogs[0].blockNumber,
```
Comment (mihoward21) [inline packages/splits-sdk/src/client/splitV2.ts:653]: ah nice, i was gonna ask why you moved this request to be afterwards. makes sense

**Review by mihoward21 — APPROVED** [review]
nice changes look good. looks like there's a merge issue? maybe just need to merge 15-94 back into this? or do a rebase and then a force push here or something?

**Comment by r0ohafza** [comment]
> nice changes look good. looks like there's a merge issue? maybe just need to merge 15-94 back into this? or do a rebase and then a force push here or something?

closing this and reopening another one with PE1594 as base

**Pull request closed without merging**